arch/arm/boot/dts/overlays/rpi-poe-overlay.dts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Overlay for the Raspberry Pi POE HAT.
+ */
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2708";
+
+	fragment@0 {
+		target-path = "/";
+		__overlay__ {
+			fan0: rpi-poe-fan@0 {
+				compatible = "rpi-poe-fan";
+				firmware = <&firmware>;
+				cooling-min-state = <0>;
+				cooling-max-state = <3>;
+				#cooling-cells = <2>;
+				cooling-levels = <0 50 150 255>;
+				status = "okay";
+			};
+		};
+	};
```

Comment:
It would be nice to have a binding document for this driver like Documentation/devicetree/bindings/hwmon/pwm-fan.txt

and a hwmon documentation like Documentation/hwmon/pwm-fan